src/misc/programs.cpp

```diff
@@ -62,6 +62,25 @@ using comdata_t = std::vector<uint8_t>;
 static std::vector<comdata_t> internal_progs_comdata;
 static std::vector<PROGRAMS_Creator> internal_progs;
 
+static uint8_t last_written_character = '\n';
+constexpr int WRITEOUT_BUFSIZE = 2048;
+
+static void WriteToStdout(std::string_view output)
+{
+	dos.internal_output = true;
+	for (const auto& chr : output) {
+		uint8_t out;
+		uint16_t bytes_to_write = 1;
+		if (chr == '\n' && last_written_character != '\r') {
+			out = '\r';
+			DOS_WriteFile(STDOUT, &out, &bytes_to_write);
+		}
+		last_written_character = out = static_cast<uint8_t>(chr);
```

Comment:
can you put the two assignments on separate lines? The second assignment can go unnoticed when skimming the code (I realize this was from the original code).

Reply:
+1, let's avoid this style